### ISSUE TITLE
Line routing

### ISSUE THREAD

**Issue by amolenaar** [gaphor/gaphas#360]
Many tools have an option to route lines, or at least create lines in a way that avoids other objects.

## Sources

To create some sort of auto-layout thingy on top of this we'll need something like COLA (Constraint Layout)(https://www.adaptagrams.org/)

* [LibAvoid](https://github.com/mjwybrow/adaptagrams) looks also very promising

Javascript implementation:  https://ialab.it.monash.edu/webcola/ (github: https://github.com/tgdwyer/WebCola)

**Comment by bollwyvl** [comment]
Yes, having robust auto layout strategies seems like table stakes for any box-and-line diagramming needs. [elk](https://eclipse.dev/elk/) might be the most widely-used open source routing algorithm toolkit out there, and handles some of the tricky stuff like ports.

Its original, EPL-2.0  implementation is in [java](https://github.com/eclipse-elk/elk), with generated bindings for a number of languages, the most widely used being [elkjs](https://github.com/kieler/elkjs), generated with the venerable closure compiler. While kinda architecturally crazy, spinning up a v8 isolate (with `mini-racer` or `pyduktape`) and handing JSON back and forth might be the most reliable, in-process way to get these benefits today.

There is a month-old, pure python port, but has a bit of the vibe-coded smell to it, and operates entirely on untyped `dict`s... might be worth revisiting if it grows some better DX, and more performant baseline characteristics.
